### PR TITLE
Adding ability to sort by hidden columns

### PR DESCRIPTION
**Pull request by marktalbot**
Hi Freek,

I've made some modifications which should take care of this described bug: https://github.com/spatie/vue-table-component/issues/14

Apologies if anything is out of order in my pull request–this is my first contribution to an open source project so this will be a learning experience for me.

Thanks for your time!

Mark

### CODE REVIEW

File: src/components/TableColumnHeader.vue [hunk 1]
```diff
@@ -48,13 +49,18 @@
                 return `table-component__th--sort-${this.sort.order}`;
             },
 
+            isVisible() {
+                return !this.column.properties.hidden;
```
Comment (sebastiandedeyne) [inline src/components/TableColumnHeader.vue]: Needs a space after the `!`

File: src/components/TableColumnHeader.vue [hunk 2]
```diff
@@ -48,13 +49,18 @@
...
+            },
+
             label() {
                 if (this.column.properties.label === null) {
                     return this.column.properties.show;
                 }
                 
                 return this.column.properties.label;
             },
+
```
Comment (sebastiandedeyne) [inline src/components/TableColumnHeader.vue]: This line of whitespace can be removed

File: src/components/TableRow.vue [hunk 3]
```diff
@@ -1,17 +1,24 @@
...
     </tr>
 </template>
 
 <script>
     export default {
         props: ['columns', 'row'],
 
+        computed: {
+            visibleColumns() {
+                return this.columns.filter(column => !column.properties.hidden);
```
Comment (sebastiandedeyne) [inline src/components/TableRow.vue]: Space after `!`

File: src/components/TableRow.vue [hunk 4]
```diff
@@ -1,17 +1,24 @@
...
+            },
+        },
+
         methods: {
             getValue(column) {
                 return this.row.getValue(column.properties.show);
             },
         },
+
```
Comment (sebastiandedeyne) [inline src/components/TableRow.vue]: Whitespace can be removed

File: src/components/TableComponent.vue [hunk 5]
```diff
@@ -151,7 +151,7 @@
             this.columns = this.$slots.default
                 .filter(column => column.componentInstance)
                 .map(column => pick(column.componentInstance, [
-                    'show', 'label', 'dataType', 'sortable', 'sortBy', 'filterable', 'filterOn',
+                    'show', 'label', 'dataType', 'sortable', 'sortBy', 'filterable', 'filterOn', 'hidden',
```
Comment (sebastiandedeyne) [inline src/components/TableComponent.vue]: No trailing comma since it's all on one line (was wrong previously)

**Comment by sebastiandedeyne** [comment]
Perfect, thanks!

**Comment by sebastiandedeyne** [comment]
Released in `v1.1.2`

**Comment by benwiley4000** [comment]
Thanks @marktalbot!

**Pull request merged**